src/Features/Core/Portable/FindUsages/IRemoteFindUsagesService.cs

```diff
@@ -92,54 +91,40 @@ public FindUsagesServerCallback(Solution solution, IFindUsagesContext context)
             _context = context;
         }
 
-        public ValueTask AddItemsAsync(int count)
-            => _context.ProgressTracker.AddItemsAsync(count);
+        public ValueTask AddItemsAsync(int count, CancellationToken cancellationToken)
+            => _context.ProgressTracker.AddItemsAsync(count, cancellationToken);
 
-        public ValueTask ItemCompletedAsync()
-            => _context.ProgressTracker.ItemCompletedAsync();
+        public ValueTask ItemCompletedAsync(CancellationToken cancellationToken)
+            => _context.ProgressTracker.ItemCompletedAsync(cancellationToken);
 
-        public ValueTask ReportMessageAsync(string message)
-            => _context.ReportMessageAsync(message);
+        public ValueTask ReportMessageAsync(string message, CancellationToken cancellationToken)
+            => _context.ReportMessageAsync(message, cancellationToken);
 
         [Obsolete]
-        public ValueTask ReportProgressAsync(int current, int maximum)
-            => _context.ReportProgressAsync(current, maximum);
+        public ValueTask ReportProgressAsync(int current, int maximum, CancellationToken cancellationToken)
+            => _context.ReportProgressAsync(current, maximum, cancellationToken);
 
-        public ValueTask SetSearchTitleAsync(string title)
-            => _context.SetSearchTitleAsync(title);
+        public ValueTask SetSearchTitleAsync(string title, CancellationToken cancellationToken)
+            => _context.SetSearchTitleAsync(title, cancellationToken);
 
-        public async ValueTask OnDefinitionFoundAsync(SerializableDefinitionItem definition)
+        public async ValueTask OnDefinitionFoundAsync(SerializableDefinitionItem definition, CancellationToken cancellationToken)
         {
-            try
-            {
-                var id = definition.Id;
-                var rehydrated = await definition.RehydrateAsync(_solution, _context.CancellationToken).ConfigureAwait(false);
-
-                lock (_idToDefinition)
-                {
-                    _idToDefinition.Add(id, rehydrated);
-                }
+            var id = definition.Id;
+            var rehydrated = await definition.RehydrateAsync(_solution, cancellationToken).ConfigureAwait(false);
 
-                await _context.OnDefinitionFoundAsync(rehydrated).ConfigureAwait(false);
-            }
-            catch (OperationCanceledException oce) when (oce.CancellationToken == _context.CancellationToken)
```

Comment:
an example of wonkyness that we had code to try to be resilient around.  this should be unnecessary now as tehre are no shenanigans around CTs.